src/core/libmaven/mzSample.cpp

```diff
@@ -1128,25 +1136,25 @@ EIC* mzSample::getEIC(float precursorMz,
 
     for (unsigned int i = 0; i < scans.size(); i++) {
         Scan* scan = scans[i];
-        if (!(scan->filterLine == filterline || filterline == ""))
+        if (filterline != "" && scan->filterLine != filterline)
             continue;
         if (scan->mslevel != 2)
             continue;
-        if (precursorMz && abs(scan->precursorMz - precursorMz) > amuQ1)
+        if (precursorMz != 0.0f && abs(scan->precursorMz - precursorMz) > amuQ1)
+            continue;
+        if (productMz != 0.0f && abs(scan->productMz - productMz) > amuQ3)
             continue;
-        // if (productMz && abs(scan->productMz - productMz) > amuQ3)
-        //	continue;
-        // if (collisionEnergy && abs(scan->collisionEnergy-collisionEnergy) >
-        // 0.5) continue;
+        if (collisionEnergy != 0.0f
+            && abs(scan->collisionEnergy - collisionEnergy) > 0.5) {
+            continue;
+        }
 
         float eicMz = 0;
         float eicIntensity = 0;
 
         switch ((EIC::EicType)eicType) {
         case EIC::MAX: {
             for (unsigned int k = 0; k < scan->nobs(); k++) {
-                if (abs(productMz - scan->mz[k]) > amuQ3)
```

Comment:
From what I remember, we added this check for every observed mass in a scan to get a fuller EIC for the selected productMz range. The productMz extracted at #358 for a scan is the nominal mass value, I'm not sure how it's calculated. Even if that value does not fall within our search window, the individual values within the scan often do. Let me know what you think.

Reply:
I tried out mzML sample files from two different MRM experiments. Both had the same kind of encoding (as `chromatogramList`). But since you say that they can also be encoded as `spectrumList`, maybe then the above might be true.

At least for data read from chromatogram lists, each single RT-intensity point gets a whole scan for itself. So that for loop (line no. 1157) is only iterating on one single observation. And therefore whether we check for productMz before or later in the loop does not matter.

Reply:
Got it. Run a check once with both structures. If the obs->mz is exactly the same as productMz then moving it out of the loop works. I suspect there would be some precision difference in the two values though, even if it's a single observation.

Reply:
@saifulbkhan hey, has the precision been verified between productMz and individual obs->mz ?

Reply:
@shubhra-agrawal If you see line no. 406 and line no. 412, the same `productMz` value is being assigned to `scan->productMz` and the scan's (one and only) observation. So our check comparing this method's `productMz` argument against the current scan's `productMz` would be the same as comparing argument with individual observation. Am I making sense? Or did I misunderstand your question?

Reply:
@saifulbkhan Oh I had missed the part where it was getting assigned to the scan's observation. Concern resolved 👍